stl/inc/algorithm

```diff
@@ -10156,6 +10146,77 @@ namespace ranges {
                 return _Found;
             }
         }
+
+    private:
+        template <class _It, class _Se, class _Pr, class _Pj>
+        _NODISCARD static constexpr minmax_result<iter_value_t<_It>> _Minmax_fwd_unchecked(
+            _It _First, const _Se _Last, _Pr _Pred, _Pj _Proj) {
+            _STL_INTERNAL_STATIC_ASSERT(forward_iterator<_It>);
+            _STL_INTERNAL_STATIC_ASSERT(sentinel_for<_Se, _It>);
+            _STL_INTERNAL_STATIC_ASSERT(indirect_strict_weak_order<_Pr, projected<_It, _Pj>>);
+
+            _STL_INTERNAL_CHECK(_First != _Last);
+
+            using _Vty = iter_value_t<_It>;
+#if _USE_STD_VECTOR_ALGORITHMS
+            if constexpr (is_same_v<_Pj, identity> && _Is_min_max_optimization_safe<_It, _Pr>
+                          && sized_sentinel_for<_Se, _It>) {
+                if (!_STD is_constant_evaluated()) {
+                    const auto _First_ptr = _STD to_address(_First);
+                    const auto _Last_ptr  = _First_ptr + (_Last - _First);
+                    const auto _Result    = _CSTD __std_minmax_element(_First_ptr, _Last_ptr);
+                    return {static_cast<_Vty>(*_Result.first), static_cast<_Vty>(*_Result.second)};

```

Comment:
Why do we use `static_cast<_Vty>(*_Result.first)` instead of `*_Result.first`.
I mean I used it too because the previous implementation had it. 
But now at libc++ review of my patch they asked me `What are these static_casts for?`

Reply:
We get here from one of two places: the `initializer_list` overload of `operator()` defined on 10084, or the `range` overload defined on 10096. If it's the former, we know that the value type of the range `_Vty` is `copyable` and the reference type is `const _Vty&` so `*_Result.first` implicitly converts to `_Vty` - we don't need the `static_cast`.

In the latter case, `indirectly_copyable_storable<_It, _Vty*>` (https://eel.is/c++draft/alg.req.ind.copy#2) also requires that `_Vty` is `copyable`, but the reference type of the input range (and `_It`) isn't necessarily `const _Vty&` so we don't have the same guarantee of implicit convertibility. In fact there's no requirement on that algorithm that lets us implicitly convert the result of an iterator dereference to `_Vty`. There _is_ a requirement that `constructible_from<_Vty, iter_reference_t<_It>>` holds, however, so we can perform the conversion explicitly with `static_cast`.